### PR TITLE
test: run Selenium tests on more browsers/JS configurations

### LINKED ISSUE #143: Re-running Selenium on Firefox and no-JS
It would be good to run the Selenium tests on Firefox _and_ Chrome, as well as these browsers with JS disabled.

### PR DESCRIPTION
**Pull request by Restioson**
By passing the browser type and whether JS is enabled via environment variables, we can run the Selenium front-end tests on all combinations of targeted browsers and JS/no-JS in the CI pipeline. The way it works from the CI perspective is by using a Matrix strategy which is then substituted into an environment variable when calling `manage.py test --tag=selenium`

Fixes #143. However, #145 is not addressed, as this PR doesn't tackle Safari. This is because GitHub Actions MacOS workers don't support  `services`, which  we use to run the postgres database. This can be left for a follow-up PR.

### CODE REVIEW

**Review by friedelwolff — COMMENTED** [review]
First commit message: suppose -> expose/support?

The text wrapping of your commit message is odd - lines starting with spaces. 

If you can reference the issue for macOS, it would be nice.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -60,14 +60,51 @@ jobs:
...
+          sudo chown runner:runner /logging
+      - name: Run Selenium Tests
+        run: |
+          cp .env.testing app/.env
+          cd app/
+          mkdir -p static_files
+          python manage.py test --tag=selenium
+        env:
+          DJANGO_SETTINGS_MODULE: app.settings
+          DATABASE_URL: postgres://sadilar:sadilar@localhost:5432/test_db
```
Comment (friedelwolff) [inline .github/workflows/testing.yml:113]: The repetition is a bit unfortunate. One solution is to combine everything before the last step and label with a yaml anchor or two to allow reuse, but that takes us towards the parts of yaml that is less readable and probably not as well known.

Another possibility is to combine all the consecutive shell commands (`pip install` up to `mkdir -p static_files`) into one step. See

**Review by friedelwolff — CHANGES_REQUESTED** [review]

File: .github/workflows/testing.yml (hunk 1 above)
Comment (Restioson) [reply .github/workflows/testing.yml:113]: I very much agree about the repetition. Unfortunately, I haven't been able to find a better way to do it yet. It looks like yaml anchors are unfortunately not supported by GitHub actions: https://github.com/actions/runner/issues/1182

**Comment by Restioson** [comment]
> First commit message: suppose -> expose/support?

I'm not quite sure what you mean? The commit message I have is "this prepares the ground for Selenium tests to be run in a Matrix strategy with multiple different configurations (with JS, without JS, Firefox, Chrome, etc)."

> The text wrapping of your commit message is odd - lines starting with spaces.

I noticed this and updated it locally a while back - seems like it got overwritten by something in the meanwhile though. I'll make sure to update it when squashing up the PR :+1:

**Review by friedelwolff — APPROVED** [review]

**Pull request merged**